app/support/order_row_importer.rb

```diff
@@ -70,10 +70,14 @@ def order_key
     @order_key ||= [user_field, chart_string_field, order_date_field]
   end
 
-  def row_with_errors # TODO: refactor
-    new_row = @row.dup
+  def row_with_errors
+    # Start with a hash of HEADERS keys with nil values to ensure optional columns
+    # are included in the report even if they are not in the uploaded CSV.
```

Comment:
Once upon a time I wrote a Hash extension that would do this cleanly called like `Hash#key_map` or something. 

Or...

```
HEADERS.invert.transform_values { |v| nil }
```